Fixes 18555: Fix model URL generator for plugins

alehaa opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #18555

<!--
    Please include a summary of the proposed changes below.
-->

The default `get_absolute_url()` implementation can now return the URL for plugins.

alehaa committed:
Fix model URL generator for plugins

alehaa committed:
Fix reverse accessor warning

jeremystretch reviewed (changes requested):
No issues with the fix itself, but I'd like to see if we can find a less disruptive solution for testing.

alehaa commented on netbox/netbox/tests/dummy_plugin/models.py:
```diff
@@ -1,7 +1,9 @@
 
-class DummyModel(models.Model):
+
+class DummyModel(NetBoxModel):
```
I don't think it's that easy, because NetBox has some logic during plugin startup. In particular, it changes the URL of the model view and the associated name. A quick local test made it crash immediately due to improper setup of the django apps logic.

Would it be okay to add an extra `DummyNetBoxModel` instead of changing the existing one?

alehaa replied:
I've gone ahead and split the test models into two separate ones.

alehaa committed:
Revert "Fix reverse accessor warning"

This reverts commit f07642bb997910205d0acc09fc8df3dde30c7981.

alehaa committed:
Add URL test case for regular models

alehaa committed:
Split dummy models

Instead of using a single model for testing, one is used for testing the
plugin API and a dedicated one is used for testing the NetBox plugin
model features.

alehaa committed:
Fix filterset test case error

jeremystretch committed:
Rename test module

jeremystretch reviewed (approved):
LGTM, thanks @alehaa! (I renamed the test module to `test_models.py` just to be consistent with the other apps.)
